File: frontend/src/lib/telegram.ts
/** Telegram 绑定 API 客户端。 */
import { api } from './auth';

export interface TelegramAccount {
  id: string;
  chat_id: number;
  username: string | null;
  enabled: boolean;
}

export interface TelegramLink {
  code: string;
  deep_link: string; // https://t.me/<bot>?start=<code>
  expires_in: number;
}

export function getTelegramAccount(): Promise<TelegramAccount | null> {
  return api.get<TelegramAccount | null>('/telegram/account');
}

export function createTelegramLink(): Promise<TelegramLink> {
  return api.post<TelegramLink>('/telegram/link-code');
}

export function unlinkTelegram(): Promise<null> {
  return api.delete<null>('/telegram/account');
}

File: backend/app/schemas/telegram.py
"""Telegram 绑定相关 Pydantic 模型。"""

import uuid

from pydantic import BaseModel, ConfigDict


class TelegramLinkOut(BaseModel):
    """生成的一次性绑定深链。"""

    code: str
    deep_link: str  # https://t.me/<bot>?start=<code>
    expires_in: int


class TelegramAccountOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    chat_id: int
    username: str | None
    enabled: bool
